Frontend/ui-driver/src/Services/Backend.purs

```diff
@@ -294,17 +298,21 @@ makeStartRideReq otp lat lon ts = StartRideReq {
 }
 --------------------------------- endRide ---------------------------------------------------------------------------------------------------------------------------------
 
-endRide :: String -> EndRideReq -> FlowBT String EndRideResponse
+endRide :: String -> EndRideReq -> Flow GlobalState (Either ErrorResponse EndRideResponse)
 endRide productId payload = do
-        headers <-getHeaders' "" false
-        withAPIResultBT (EP.endRide productId) identity errorHandler (lift $ lift $ callAPI headers (EndRideRequest productId payload))
+        headers <- getHeaders "" false
+        withAPIResult (EP.endRide productId) unwrapResponse $ callAPI headers ((EndRideRequest productId payload))
     where
-      errorHandler (ErrorPayload errorPayload) =  do
-            void $ lift $ lift $ toggleLoader false
-            BackT $ pure GoBack
+      unwrapResponse (x) = x
+
 
-makeEndRideReq :: Number -> Number -> Maybe Boolean -> Int -> Int -> String -> EndRideReq
-makeEndRideReq lat lon numDeviation tripDistance tripDistanceWithAcc ts = EndRideReq {
+makeEndRideReq :: Maybe String -> Maybe String -> Maybe String -> Number -> Number -> Maybe Boolean -> Int -> Int -> String -> EndRideReq
+makeEndRideReq endOtp endOdometerReading fileId lat lon numDeviation tripDistance tripDistanceWithAcc ts = EndRideReq {
+    "endRideOtp" : endOtp,
+    "odometer" : (\value -> Odometer {
+        "value" : fromMaybe 0.0 $ Number.fromString value,
```

Comment:
we are doing that only, if endOdometerReading is nothing we are sending nothing,
here we are using converting it `fromMaybe 0.0 $ Number.fromString value,` 
should we make "odometer" nothing if fromString gives nothing ?

Reply:
I think it's like this because we are calling `fromString` which again returns maybe, considering this we can re-write this where 0.0 should not go even if `fromString` function fails. But, It's okay for now, we can look into major fixes majorly